x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/DelegatePkiAuthenticationRequest.java

```diff
@@ -18,29 +23,68 @@
 import java.security.cert.CertificateException;
 import java.security.cert.CertificateFactory;
 import java.security.cert.X509Certificate;
-import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
 
 import static org.elasticsearch.action.ValidateActions.addValidationError;
 
-public final class DelegatePkiAuthenticationRequest extends ActionRequest {
+/**
+ * The request object for {@code TransportDelegatePkiAuthenticationAction} containing the certificate chain for the target subject
+ * distinguished name to be granted an access token.
+ */
+public final class DelegatePkiAuthenticationRequest extends ActionRequest implements ToXContentObject {
+
+    private static final ParseField X509_CERTIFICATE_CHAIN_FIELD = new ParseField("x509_certificate_chain");
+
+    public static final ConstructingObjectParser<DelegatePkiAuthenticationRequest, Void> PARSER = new ConstructingObjectParser<>(
+            "delegate_pki_request", false, a -> {
+                @SuppressWarnings("unchecked")
+                final List<String> encodedCertificatesList = (List<String>) a[0];
+                final List<X509Certificate> certificates = new ArrayList<>(encodedCertificatesList.size());
+                try {
+                    final CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+                    for (int i = 0; i < encodedCertificatesList.size(); i++) {
+                        try (ByteArrayInputStream bis = new ByteArrayInputStream(
+                                Base64.getDecoder().decode(encodedCertificatesList.get(i)))) {
+                            certificates.add((X509Certificate) certificateFactory.generateCertificate(bis));
+                        } catch (CertificateException | IOException e) {
+                            throw new RuntimeException(e);
+                        }
+                    }
+                } catch (CertificateException e) {
+                    throw new RuntimeException(e);
+                }
+                return new DelegatePkiAuthenticationRequest(certificates);
+            });
+
+    static {
+        PARSER.declareStringArray(ConstructingObjectParser.constructorArg(), X509_CERTIFICATE_CHAIN_FIELD);
```

Comment:
It's a personal preference, but I would have implemented this with `declareFieldArray` so that you could simply provide a `String` to `X509Certificate` parser:
```
PARSER.declareFieldArray(ConstructingObjectParser.constructorArg(),  
   (parser, c) -> parseCertificate(parser.text()), 
   X509_CERTIFICATE_CHAIN_FIELD, ValueType.STRING_ARRAY);
```

Reply:
Neat! thanks for the suggestion.